src/Button.tsx

```diff
@@ -0,0 +1,241 @@
+   *
+   * Useful for Storybook
+   *
+   * @ignore
+   */
+  forceHoverState?: boolean;
+
+  /**
+   * Either an icon to show to the left of the button text, or on it's own
+   */
+  icon?: React.ReactElement;
+
+  /**
+   * Size of the button
+   *
+   * Defaults to "default"
+   */
+  size?: "default" | "small" | "large";
+
+  /**
+   * Which variant to display
+   *
+   * The options are as follows:
+   *
+   * - `normal`: A button with a border and a background
+   * - `simple`: No border or background (overrideable).
+   *     - This will require you adding an `:active` style in addition to `:hover`
+   */
+  variant?: "normal" | "simple";
+}
+
+/**
+ * Style system for Space Kit buttons
+ *
+ * This is intended to be used as an abstraction for your project's style guide.
+ * There are way too many configuration options to use this component directly.
+ *
+ * @see https://zpl.io/amdN6Pr
+ */
+export const Button: React.FC<Props> = ({
+  children,
+  disabled = false,
+  fab = false,
+  forceActiveState = false,
+  forceFocusState = false,
+  forceHoverState = false,
+  icon,
+  size = "default",
+  variant = "normal",
+  ...otherProps
+}) => {
+  /**
+   * Icon size in pixels
+   *
+   * This is stored so we can use the same value for `height` and `width`
+   */
+  const iconSize = size === "small" ? 12 : size === "large" ? 24 : 16;
+
+  const iconOnly = !children;
+
+  if (fab) {
+    if (!icon) {
+      throw new TypeError("FAB buttons are required to have an `icon`");
+    } else if (children) {
+      throw new TypeError("FAB buttons cannot have children, only an `icon`");
+    }
+  }
+
+  if (forceActiveState && forceHoverState) {
+    throw new TypeError(
+      "Do not force multiple properties at once, you will get indeterminiate behavior"
+    );
+  }
+
+  /**
+   * Properties applied to for an active state.
+   *
+   * We store this so we can use it either in an `:active` style or when
+   * `forceActiveState` is set.
+   */
+  const activeProperties = {
+    // The `box-shadow` property is copied directly from Zeplin
+    boxShadow:
+      "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)",
+    outline: 0,
+  };
+
+  return (
+    <button
+      {...otherProps}
+      disabled={disabled}
+      {...!disabled && forceActiveState && { "data-force-active-state": true }}
+      {...!disabled && forceFocusState && { "data-force-focus-state": true }}
+      {...!disabled && forceHoverState && { "data-force-hover-state": true }}
+      css={{
+        "&[disabled]": {
+          backgroundColor: colors.silver.dark,
+          color: colors.grey.light,
+
+          // We need to also set the `:hover` on `:disabled` so it has a higher
+          // specificity than any `:hover` classes passed in. This also means
+          // that both of these need to be overriden if we want to use a custom
+          // disabled color.
+          ":hover": {
```

Comment:
Alternatively, we could define non-disabled hover styles like this:

```js
{
  ':not([disabled]):hover': { // this selector might need a & before :not
    // hover styles here
  }
}
```

Reply:
I wanted to avoid complicated css Nesting here because it would be impossible to represent overrides for this with tailwindcss, which would make this library less extensible.